File: Running Sum of 1d Array - Leetcode 1480/Running Sum of 1d Array - Leetcode 1480.js
var runningSum = function(nums) {
    var s = 0;
    var n = nums.length;
    var prefix_sum = new Array(n).fill(0);
    
    for (var i = 0; i < n; i++) {
        s += nums[i];
        prefix_sum[i] = s;
    }
    
    return prefix_sum;
    // Time: O(n)
    // Space: O(n)
};

File: Running Sum of 1d Array - Leetcode 1480/Running Sum of 1d Array - Leetcode 1480.java
class Solution {
    public int[] runningSum(int[] nums) {
        int s = 0;
        int[] running_sum = new int[nums.length];
        
        for (int i = 0; i < nums.length; i++) {
            s += nums[i];
            running_sum[i] = s;
        }
        
        return running_sum;
    }
    // Time: O(n)
    // Space: O(n)
}
